### PR TITLE
KIALI-1650 Delete Istio-config objects

### PR DESCRIPTION
**Pull request by jotak**
Supersedes #533 
Thanks to @pilhuhn for the original PR

JIRA: https://issues.jboss.org/browse/KIALI-1650

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#193](https://paste.fedoraproject.org/paste/Dj3mZhSH~aoJMnDEGyN0MA)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1627]](https://paste.fedoraproject.org/paste/h1SyckWCyX1~DrhCdG-IIw)
  * :heavy_check_mark: Istio-Upstream [#[475]](https://paste.fedoraproject.org/paste/f4NoLRHYuUENKS6LjykB7w)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1820, 1821]](https://paste.fedoraproject.org/paste/Ugz~zYWsX03e~Zg9o8uMhQ)
  * :heavy_check_mark: install-bookinfo [#[1002]](https://paste.fedoraproject.org/paste/Tvyf65rZTAykVVhcrFQSZg)
  * :heavy_check_mark: Kiali-on-Openshift [#[1018]](https://paste.fedoraproject.org/paste/hQ11S2KnkxZM6w9PKDew4Q)
  * :red_circle: run-kiali-e2e-tests [#[424]](https://paste.fedoraproject.org/paste/vaKDvMZ1IW8xxLEIy7oLuQ)


KIALI_NAME=kialicorepr566-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#194](https://paste.fedoraproject.org/paste/Ar68t~KtBi-RIxIZuEpJxw)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1629]](https://paste.fedoraproject.org/paste/BtT11vPCYR7L8daxYW02fA)
  * :heavy_check_mark: Istio-Upstream [#[477]](https://paste.fedoraproject.org/paste/PSN46Gxgmr6iRAz~NZtSbQ)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1824, 1825]](https://paste.fedoraproject.org/paste/vJsTPnsMZQQWMPM8nE1YAA)
  * :heavy_check_mark: install-bookinfo [#[1004]](https://paste.fedoraproject.org/paste/8SqGy5tEWEw1995-CYN78w)
  * :heavy_check_mark: Kiali-on-Openshift [#[1020]](https://paste.fedoraproject.org/paste/wB~araiyFzJ85GTwPlYQEQ)
  * :red_circle: run-kiali-e2e-tests [#[426]](https://paste.fedoraproject.org/paste/woNwi-mzpCyVn3YOFksqhA)


KIALI_NAME=kialicorepr566-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#246](https://paste.fedoraproject.org/paste/hPmAxKG0gmghOs45Tev8~Q)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1749]](https://paste.fedoraproject.org/paste/cTmgulDwiWSHA-C8nsjHjA)
  * :heavy_check_mark: Istio-Upstream [#[594]](https://paste.fedoraproject.org/paste/OhbTkOLJXyJb~zfTheo~Eg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2042, 2043]](https://paste.fedoraproject.org/paste/AgS9oUcdpyJ9zUS4KOFe2g)
  * :heavy_check_mark: install-bookinfo [#[1122]](https://paste.fedoraproject.org/paste/nbTHX-WjOyEV~c5yTlZhXw)
  * :heavy_check_mark: Kiali-on-Openshift [#[1114]](https://paste.fedoraproject.org/paste/Svc-GaYULkjpfqbwfit3Gw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[522]](https://paste.fedoraproject.org/paste/kihv91zAR-qt8O6eg9ECVQ)


KIALI_NAME=kialicorepr566-istio-system

File: handlers/istio_config.go [hunk 1]
```diff
@@ -121,14 +121,43 @@ func IstioConfigDetails(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	// Get business layer
+	business, err := business.Get()
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, "Services initialization error: "+err.Error())
+		return
+	}
+
+	err = business.IstioConfig.DeleteIstioConfigDetail(api, namespace, objectType, object)
```
Comment (lucasponce) [inline handlers/istio_config.go]: Question, if object doesn't exist. This will send an OK or an 404 ?

Comment (lucasponce) [reply handlers/istio_config.go]: I mean, perhaps it's just a matter to check if there is an 404 in the err that comes from the underlying API and just change this to not return always a 500.

Comment (jotak) [reply handlers/istio_config.go]: you're right i'll check that

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#249](https://paste.fedoraproject.org/paste/FMq5zgM~owVzXo8lY~yjqA)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1756]](https://paste.fedoraproject.org/paste/90nJ0TBLqbSoFGEHbvKv5w)
  * :heavy_check_mark: Istio-Upstream [#[601]](https://paste.fedoraproject.org/paste/eM84XT6xfspVRHCwubA1oA)
  * :black_circle: Kiali-Istio-Checker [#[2056, 2058]](https://paste.fedoraproject.org/paste/mCBk7i7ZJ-Mf-bBfqbxnIg)
  * :heavy_check_mark: install-bookinfo [#[1131]](https://paste.fedoraproject.org/paste/NPm1h2LDn5QLkmn0fDYPRw)
  * :red_circle: Kiali-on-Openshift [#[1121]](https://paste.fedoraproject.org/paste/F7arwIYAVzw5lKvbmm835A)


KIALI_NAME=kialicorepr566-istio-system

**Pull request merged**